pkg/client/client.go

```diff
@@ -152,7 +152,23 @@ func (c *Client) callAPI(request *http.Request) (*http.Response, error) {
 		log.Printf("\n%s\n", string(dump))
 	}
 
-	resp, err := c.cfg.HTTPClient.Do(request)
+	attempts := 1
+	if request.Method == http.MethodGet {
+		attempts += c.cfg.GetRequestRetries
+	}
+
+	for i := attempts; i > 0; i-- {
+		resp, err = c.cfg.HTTPClient.Do(request)
```

Comment:
This retry logic is pretty basic.  I wonder if this would be enough for someone who wanted to retry more than once.  I've noticed that go libraries often allow you to pass in a custom http client which could handle the retries for you.  For example, when I added retries to the metronome client, I was able to just pass in retryable_http client which has exponential backoff and some other nice features built into it: https://github.com/cockroachlabs/managed-service/blob/master/console/metronome/client.go#L1129

I'm on board with keeping this basic version but I feel like we should also allow passing in a custom client and potentially mention in the read me that "more complex retry logic can be attained by passing in a custom retry logic such as... "